src/query/functions/aggregation/function.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package aggregation
+
+import (
+	"math"
+)
+
+// SumType adds all non nan elements in a list of series
+const SumType = "sum"
+
+func sumFn(values []float64, indices [][]int) []float64 {
+	sumByIndices := make([]float64, len(indices))
+	for i, indexList := range indices {
+		sum := 0.0
+		for _, idx := range indexList {
+			v := values[idx]
+			if !math.IsNaN(v) {
+				sum += v
+			}
+		}
+
+		sumByIndices[i] = sum
+	}
+
+	return sumByIndices
+}
+
+// MinType takes the minimum all non nan elements in a list of series
+const MinType = "min"
+
+func minFn(values []float64, indices [][]int) []float64 {
+	minByIndices := make([]float64, len(indices))
+	for i, indexList := range indices {
+		min := math.NaN()
```

Comment:
Is this logic straight from Prom?

Reply:
Not sure what you mean; I'm adding the functions from 
https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators but using my own logic for them